src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -136,14 +136,19 @@ public AsymmetricSignatureProvider(SecurityKey key, string algorithm, bool willC
             _cryptoProviderFactory = key.CryptoProviderFactory;
             _minimumAsymmetricKeySizeInBitsForSigningMap = new Dictionary<string, int>(DefaultMinimumAsymmetricKeySizeInBitsForSigningMap);
             _minimumAsymmetricKeySizeInBitsForVerifyingMap = new Dictionary<string, int>(DefaultMinimumAsymmetricKeySizeInBitsForVerifyingMap);
+
+            var jsonWebKey = key as JsonWebKey;
```

Comment:
Perhaps it's a bit cleaner to move this code to ResolveAsymmetricAdapter method.

Reply:
Actually, I'm not sure that we really need this code.
If a key is JsonWebKey, it will be converted by AsymmetricAdapter.

Reply:
Nevermind, we need the converted key down the stack when key size validation is done.